### PR TITLE
Add link to obtain a letter as TEI/XML

### LINKED ISSUE #76: TEI/XML export
Add link to download a letter as TEI/XML

using this configuration: https://github.com/dhlab-basel/Knora/tree/develop/webapi/_test_data/test_route/texts/beol

figure out how to embed referred resources: type, label, Iri

### PR DESCRIPTION
**Pull request by tobiasschweizer**
closes #76

### CODE REVIEW

**Review by subotic — APPROVED** [review]
LGTM, thanks :-)

**Comment by tobiasschweizer** [comment]
@SepidehAlassi I am thinking about making the TEI link generation code a directive.

**Comment by SepidehAlassi** [comment]
@tobiasschweizer that is a good idea!

**Comment by tobiasschweizer** [comment]
@SepidehAlassi I made it a directive. It only creates a TEI link if there is a proper config element.

**Comment by tobiasschweizer** [comment]
@SepidehAlassi Could you have a look at this PR?

**Comment by SepidehAlassi** [comment]
[TEI export](https://trello.com/c/8dR7x3Iu/143-tei-export)

**Comment by SepidehAlassi** [comment]
@tobiasschweizer now that you have removed the config for the letter, is there anything I else I should change for checking this PR?

**Comment by tobiasschweizer** [comment]
@SepidehAlassi you just have to provide the settings in the config (https://github.com/dhlab-basel/beol/blob/develop/src/config/config.dev.json) like in the test for the directive:

```
{
	            ...
	            'tei': {
	                'http://0.0.0.0:3333/ontology/0801/beol/v2#letter': {
	                    'textProperty': 'http://0.0.0.0:3333/ontology/0801/beol/v2#hasText',
	                    'mappingIRI': 'http://rdfh.ch/projects/yTerZGyxjZVqFMNNKXCDPF/mappings/BEOLTEIMapping',
	                    'gravsearchTemplateIri': 'http://rdfh.ch/0801/templateIri',
	                    'teiHeaderXSLTIri': 'http://rdfh.ch/0801/headerIri'
	                }
	            }
	        }
```


You get the settings when you run the Python script in Knora (https://github.com/dhlab-basel/Knora/blob/develop/webapi/_test_data/test_route/texts/beol/beolTEI.py).

**Comment by tobiasschweizer** [comment]
But actually this PR is just about generating the link and display it, so it is actually enough to look at the directive.

**Comment by SepidehAlassi** [comment]
@tobiasschweizer I see that the link is created for a letter as
`<span _ngcontent-c24="" appteilink="" ng-reflect-resource-type="http://0.0.0.0:3333/ontology/0" ng-reflect-resource-iri="http://rdfh.ch/0801/4uMWaXMsT1"></span>`
if I inspect the html, but nothing is displayed in the GUI.
I can not fully understand how is the user supposed to use this link? In one of the previous PRs, there was a clickable component which would result in the creation of TEIs, but there is nothing of that sort anymore.

**Comment by SepidehAlassi** [comment]
@tobiasschweizer I see, now all works. I had a problem with configuration, it did not show me the link.

File: src/app/resource/letter/letter.component.html [hunk 1]
```diff
@@ -148,6 +148,10 @@ <h4 class="subheading-2">{{ontologyInfo?.getLabelForProperty(props?.sysnum[0].pr
                 <read-text-value [valueObject]="props?.text[0]" [bindEvents]="true" [ontologyInfo]="ontologyInfo"></read-text-value>
             </section>
 
+            <section>
+                <span appTeiLink [resourceIri]="iri" [resourceType]="'http://0.0.0.0:3333/ontology/0801/beol/v2#letter'"></span>
+            </section>
```
Comment (SepidehAlassi) [inline src/app/resource/letter/letter.component.html]: I believe it is better to have this link displayed on the right top corner of the letter template. As it is now, the user has to scroll down to end to see the link. 
Most users like to see the things in a first glance without having to scroll.

**Review by SepidehAlassi — CHANGES_REQUESTED** [review]

File: src/app/resource/letter/letter.component.html (hunk 1 above)
Comment (tobiasschweizer) [reply src/app/resource/letter/letter.component.html]: done in https://github.com/dhlab-basel/beol/pull/135/commits/31a829fe672a482ba69d2ec84a4baad6a2c2510b

the link is now shown above the letter's text

**Comment by tobiasschweizer** [comment]
top right placement:  https://github.com/dhlab-basel/beol/pull/135/commits/ab96df08da50db9bf8f24c12ae07c2618d7fbc6a

**Review by SepidehAlassi — APPROVED** [review]

**Pull request merged**